mdsuite/calculators/coordination_number_calculation.py

```diff
@@ -109,267 +140,279 @@ def __init__(self, **kwargs):
+        peaks = self._get_rdf_peaks(rdf)  # get the max value indices
 
         # Calculate the range in which the coordination numbers should exist.
-        cn_radii_1 = golden_section_search(
-            [self.radii, self.rdf], self.radii[peaks[1]], self.radii[peaks[0]]
-        )
-        cn_radii_2 = golden_section_search(
-            [self.radii, self.rdf], self.radii[peaks[2]], self.radii[peaks[1]]
-        )
-
-        # Locate the indices of the radii values
-        cn_indices_1 = list(
-            [
-                np.where(self.radii == cn_radii_1[0])[0][0],
-                np.where(self.radii == cn_radii_1[1])[0][0],
-            ]
-        )
-        cn_indices_2 = list(
-            [
-                np.where(self.radii == cn_radii_2[0])[0][0],
-                np.where(self.radii == cn_radii_2[1])[0][0],
-            ]
-        )
+        coordination_shells = {}
+        for i in range(self.args.number_of_shells):
+            coordination_shells[i] = np.zeros(2, dtype=int)
+            cn_radii_range = golden_section_search(
+                [radii, rdf], radii[peaks[i + 1]], radii[peaks[i]]
+            )
+            for j in range(2):
+                coordination_shells[i][j] = np.where(radii == cn_radii_range[j])[0][0]
 
-        return cn_indices_1, cn_indices_2
+        return coordination_shells
 
-    def _get_coordination_numbers(self):
+    def _get_coordination_numbers(
+        self, integral_data: np.ndarray, radii: np.ndarray, rdf: np.ndarray
+    ) -> dict:
         """
         Calculate the coordination numbers
+
+        Parameters
+        ----------
+        integral_data : np.ndarray
+                Integrated RDF data from which to compute the coordination numbers.
+        radii : np.ndarray
+                radii data to use in the analysis
+        rdf : np.ndarray
+                RDF data to use in the analysis
+
+        Returns
+        -------
+        coordination_numbers : dict
+                A dictionary of coordination numbers.
         """
 
-        self.indices = self._find_minima()  # get the minimums
+        coordination_shells = self._find_minima(radii, rdf)  # get the minimums
 
-        # Calculate the coordination numbers by averaging over the two values
-        # returned by _find_minimums
-        first_shell = np.mean(
-            [
-                self.integral_data[self.indices[0][0]],
-                self.integral_data[self.indices[0][1]],
-            ]
-        )
-        first_shell_error = (
-            np.std(
-                [
-                    self.integral_data[self.indices[0][0]],
-                    self.integral_data[self.indices[0][1]],
-                ]
+        coordination_numbers = {}
+
+        for key, val in coordination_shells.items():
+            lower_bound = integral_data[val[0]]
+            upper_bound = integral_data[val[1]]
+
+            coordination_numbers[f"CN_{int(key) + 1}"] = np.mean(
```

Comment:
maybe add a function `get_key` `lamnda key: f"CN_{int(key) + 1}"` so we could potentially change how it is stored in a single place.